### ISSUE TITLE
Server (api routes) not working in new app structure 

### ISSUE THREAD

**Issue by Gobler** [nuxt/nuxt#27780]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.20.3
- Nuxt Version:     3.12.2
- CLI Version:      3.12.0
- Nitro Version:    2.9.6
- Package Manager:  pnpm@8.15.6
- Builder:          -
- User Config:      future
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-hqairf-w3szgs

### Describe the bug

In the application it is possible to use everything provided by the layer (pages, components etc), in the layer it is possible to use everything provided by the application unfortunately server api routes located in the layer do not work - it is only possible to use server api routes from the application.

### Additional context

Uses compatibility with nuxt4

**Comment by martinszeltins** [comment]
@Gobler API routes are working for me with layers (using Nuxt 4 compatability). The only difference is that I am manually extending layers in `nuxt.config.ts` instead of using the new `layers` directory. So the issue seems to be related to the new `layers` directory.

Reproduction:
https://stackblitz.com/~/github.com/martinszeltins/nuxt4-layers-server-api?file=app/events/pages/events.vue

**Comment by Gobler** [comment]
@martinszeltins If I see it correctly, the layer with events does not have compatibility set with version 4 - if you set in nuxt.config.ts to make the layer compatible as well, and use the new structure in the layer, it will stop working.

Well, and above all - the new structure is supposed to separate the individual layers from the application so adding layers in the application directory is not a good practice in my opinion.

**Comment by martinszeltins** [comment]
> @martinszeltins If I see it correctly, the layer with events does not have compatibility set with version 4 - if you set in nuxt.config.ts to make the layer compatible as well, and use the new structure in the layer, it will stop working.

Actually, Nuxt merges all layers together, so it works even if I set compatibility version 4 for events layer specifically. Still works.

> Well, and above all - the new structure is supposed to separate the individual layers from the application so adding layers in the application directory is not a good practice in my opinion.

I see. In my case the layers _**are**_ the application, meaning there is no application outside of layers, so it works fine for my case, but your case might be different.

Anyway, seems like it doesn't like the `layers` directory.

**Comment by Gobler** [comment]
> > @martinszeltins If I see it correctly, the layer with events does not have compatibility set with version 4 - if you set in nuxt.config.ts to make the layer compatible as well, and use the new structure in the layer, it will stop working.
> 
> Actually, Nuxt merges all layers together, so it works even if I set compatibility version 4 for events layer specifically. Still works.
> 
> > Well, and above all - the new structure is supposed to separate the individual layers from the application so adding layers in the application directory is not a good practice in my opinion.
> 
> I see. In my case the layers _**are**_ the application, meaning there is no application outside of layers, so it works fine for my case, but your case might be different.
> 
> Anyway, seems like it doesn't like the `layers` directory.

Right, my mistake - I was checking two things at the same time. Anyway, the underlying problem remains unsolved.

**Comment by Gobler** [comment]
@danielroe, I saw that you updated the tags.... anything more known about this error, is there any plan to fix it?

**Comment by Gobler** [comment]
Maybe it will help to solve the problem: 
If I move the `server` folder in the layer to the `app` folder  and in nuxt.config.ts set `serverDir: "app/server"` then Everything seems to work correctly - apparently something is not interpreted correctly for the new application structure

**Comment by oripka** [comment]
@Gobler I ran into the same problem and using `serverDir` fixed it

**Comment by Gobler** [comment]
> @Gobler I ran into the same problem and using `serverDir` fixed it

Yes, this fixes the problem and for the moment this is how I have dealt with it while it would be good if it worked in the default configuration ;)
